### PR TITLE
Warn users when it's unable to assign average slot

### LINKED ISSUE #2255: AttributeError: 'TrackableWeightHandler' object has no attribute 'trainable'
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 18.04
- TensorFlow version and how it was installed (source or binary): 2.3.0
- TensorFlow-Addons version and how it was installed (source or binary): 0.11.2
- Python version: 3.6.8
- Is GPU used? (yes/no): yes

**Describe the bug**

```TrackableWeightHandler``` doesn't have trainable and assign attributes but also to be appended to weights and trainable_weights. When I use ```AverageModelCheckpoint```, I meet the bug. I'm not sure it's a bug of addons or tensorflow.
```TrackableWeightHandler``` is added in ```TextVectorization```

```
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/root/.vscode-server/extensions/ms-python.python-2020.11.371526539/pythonFiles/lib/python/debugpy/__main__.py", line 45, in <module>
    cli.main()
  File "/root/.vscode-server/extensions/ms-python.python-2020.11.371526539/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 430, in main
    run()
  File "/root/.vscode-server/extensions/ms-python.python-2020.11.371526539/pythonFiles/lib/python/debugpy/../debugpy/server/cli.py", line 267, in run_file
    runpy.run_path(options.target, run_name=compat.force_str("__main__"))
  File "/usr/lib/python3.6/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/usr/lib/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/srv/automl/efficientdet/nlp.py", line 65, in <module>
    tfa.callbacks.AverageModelCheckpoint(update_weights=True, filepath=os.path.join(MODEL_DIR, 'ckpt_moving_average'))])
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training.py", line 108, in _method_wrapper
    return method(self, *args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/training.py", line 1137, in fit
    callbacks.on_epoch_end(epoch, epoch_logs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/callbacks.py", line 412, in on_epoch_end
    callback.on_epoch_end(epoch, logs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/callbacks.py", line 1249, in on_epoch_end
    self._save_model(epoch=epoch, logs=logs)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_addons/callbacks/average_model_checkpoint.py", line 76, in _save_model
    self.model.optimizer.assign_average_vars(self.model.trainable_weights)
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_addons/optimizers/average_wrapper.py", line 125, in assign_average_vars
    for var in var_list
  File "/usr/local/lib/python3.6/dist-packages/tensorflow_addons/optimizers/average_wrapper.py", line 126, in <listcomp>
    if var.trainable
AttributeError: 'TrackableWeightHandler' object has no attribute 'trainable'
```

**Code to reproduce the issue**
https://colab.research.google.com/drive/1OM7kAA6aOcNSTTXpd74kUnw-EeeM2uH5?usp=sharing

Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**

Change ```model.variables``` to ```model.trainable_weights``` could fix it temporarily.

### PR DESCRIPTION
**Pull request by WindQAQ**
# Description

Fixes #2255 

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [x] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

New tests with preprocessing layer.

### CODE REVIEW

**Comment by WindQAQ** [comment]
@fsx950223 Hi, can you take a look at the PR when time allows? Thank you!

File: tensorflow_addons/optimizers/average_wrapper.py [hunk 1]
```diff
@@ -103,14 +104,18 @@ def assign_average_vars(self, var_list):
...
-                for var in var_list
-                if var.trainable
-            ]
-        )
-        return assign_op
+        assign_ops = []
+        try:
+            for var in var_list:
+                assign_ops.append(
+                    var.assign(
```
Comment (fsx950223) [inline tensorflow_addons/optimizers/average_wrapper.py:111]: Should try-catch each assign op or all assign ops?

Comment (WindQAQ) [reply tensorflow_addons/optimizers/average_wrapper.py:111]: Ops, it's my fault. Let me correct it real quick.

Comment (WindQAQ) [reply tensorflow_addons/optimizers/average_wrapper.py:111]: Fixed. Thanks for pointing out!

**Review by fsx950223 — APPROVED** [review]

File: tensorflow_addons/optimizers/average_wrapper.py [hunk 2]
```diff
@@ -103,14 +104,17 @@ def assign_average_vars(self, var_list):
...
+        assign_ops = []
+        for var in var_list:
+            try:
+                assign_ops.append(
+                    var.assign(
+                        self.get_slot(var, "average"), use_locking=self._use_locking
+                    )
+                )
+            except Exception as e:
+                warnings.warn("Unable to assign average slot to {} : {}".format(var, e))
```
Comment (fsx950223) [inline tensorflow_addons/optimizers/average_wrapper.py:116]: It's better to use ```tf.get_logger().warning```?

Comment (WindQAQ) [reply tensorflow_addons/optimizers/average_wrapper.py:116]: Seems that other than `seq2seq`, most components in addons use `warnings.warn` print warnings. We should make it consistent.

**Review by seanpmorgan — APPROVED** [review]
LGTM thx!

**Pull request merged**